### PR TITLE
Improve error message when generator not found

### PR DESCRIPTION
**Pull request by mischah**
Closes yeoman/yo#388

Error message before:
before

Error message after:
after

### CODE REVIEW

**Comment by mischah** [comment]
I stripped the total amount of generators from the error message. Because I guess it will confuse regular users like it confused me.

Reason: `this.namespaces().length` contains subgenerators.

**Comment by mischah** [comment]
I wonder if @SBoudrias has a bot running which  adds `to review` labels on PRs automagically :flushed:

**Comment by SBoudrias** [comment]
Yes, there's a waffleboard tracking the whole projects: https://waffle.io/yeoman/generator

Can we remove the back tick from the command line. And instead of bold, let's use a color like cyan so it works on windows.

**Comment by mischah** [comment]
> Can we remove the back tick from the commands. And instead of bold, let's use a color like cyan so it works on windows.

Yup:
update

**Comment by SBoudrias** [comment]
Yeah, could use quotes.

Also what do you think about making the instruction more specific to the generator the user tried to run? e.g. `npm install generator-allo-allo`

File: lib/environment.js [hunk 1]
```diff
@@ -298,10 +298,14 @@ Environment.prototype.create = function create(namespace, options) {
...
-        chalk.bold(' and then install them with ' + 'npm install [name]') + '.\n' +
-        'To see the ' + this.namespaces().length + ' registered generators run yo with the `--help` option.'
+        chalk.red('You don\’t seem to have a generator with the name ' + chalk.bold(namespace) + ' installed.') + '\n' +
+        'But help is on the way:\n\n' +
+        'You can see available generators via ' +
+        chalk.yellow('npm search yeoman-generator') + ' or via ' + chalk.yellow('http://yeoman.io/generators/') + '. \n' +
+        'Install them with ' + chalk.yellow('npm install [generator-name]') + '.\n\n' +
+        'To see all your installed generators run ' + chalk.yellow('yo') + ' without any arguments. ' +
+        'Adding the ' + chalk.yellow('--help') + ' option will also show subgenerators. \n\n' +
+        'Run ' + chalk.yellow('yo doctor') + ' if you encounter problems with running installed generators.'
```
Comment (SBoudrias) [inline lib/environment.js]: What do you think about:

> If yo cannot find the generator, run `yo doctor` to troubleshoot your system.

**Comment by SBoudrias** [comment]
This is great, thanks a lot!!

**Comment by mischah** [comment]
My pleasure :octocat: 
Or like you native speakers like to say »You scratch my back, I scratch yours« 😄

**Comment by SBoudrias** [comment]
I'll prepare a release later today or tomorrow.

**Comment by mischah** [comment]
👍

**Pull request merged**